coalib/io/FileCache.py

```diff
@@ -0,0 +1,137 @@
+import functools
+import sys
+import os
+import tokenize
+
+__all__ = ["get_content", "clearcache", "checkcache"]
+
+
+# The cache. Maps filenames to either a thunk which will provide source code,
+# or a tuple (size, mtime, contents, fullname) once loaded.
+cache = {}
```

Comment:
The cache should be tied to a `FileFactory` object, and not stay in global scope. That makes the cache cleaning up easier because it automatically cleans up as soon as the file factories are destructed. If someone wants to keep the cache alive, the relvant  file factory objects should be kept alive.

Reply:
So do you want a `self.cache` in `FileFactory` to tie it to a Factory object?